intel-gpu-exporter.py

```diff
@@ -151,29 +176,35 @@ def update(data):
     )
 
     logging.info("Started " + cmd)
-    output = ""
-
-    if os.getenv("IS_DOCKER", False):
-        for line in process.stdout:
-            line = line.decode("utf-8").strip()
-            output += line
-
-            try:
-                data = json.loads(output.strip(","))
-                logging.debug(data)
-                update(data)
-                output = ""
-            except json.JSONDecodeError:
-                continue
-    else:
-        while process.poll() is None:
-            read = process.stdout.readline()
-            output += read.decode("utf-8")
-            logging.debug(output)
-            if read == b"},\n":
-                update(json.loads(output[:-2]))
-                output = ""
-
+    # Robust streaming JSON parse: bracket-depth framing
+    buf = ''
+    depth = 0
+    started = False
+    while True:
+        chunk = process.stdout.read(4096)
+        if not chunk:
+            break
+        for ch in chunk.decode('utf-8', 'ignore'):
+            if not started:
+                if ch == '{':
+                    started = True
+                    depth = 1
+                    buf = '{'
+            else:
+                buf += ch
+                if ch == '{':
+                    depth += 1
+                elif ch == '}':
+                    depth -= 1
+                    if depth == 0:
+                        try:
+                            data = json.loads(buf)
+                            logging.debug(data)
+                            update(data)
+                        except Exception:
+                            pass
+                        buf = ''
+                        started = False
```

Comment:
Can you revert this section of the change and test if it still works in your setup?